### PR TITLE
Temp fix

### PR DESCRIPTION
**Pull request by beylerian**
added exception nets temporarily

### CODE REVIEW

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 1]
```diff
@@ -75,9 +64,13 @@ public void onCreate(SQLiteDatabase db) {
...
-        SQLiteDatabase db = getWritableDatabase();
-        db.delete(TABLE_NAME, "", new String[]{});
-        db.close();
+        try {
+            SQLiteDatabase db = getWritableDatabase();
+            db.delete(TABLE_NAME, "", new String[]{});
+            db.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: I think when we get exception, there are some cases where the close() not called.
so how about 
add finally close and close() it there.

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 2]
```diff
@@ -125,20 +109,16 @@ public void updateOrInsert(long dateAndHour, int stepsSinceBoot) {
...
-                db.endTransaction();
-            }
-        } catch (SQLiteDatabaseLockedException | IllegalStateException e) {
-            Crashlytics.logException(e);
+            updateOrInsertWithoutTransaction(db, dateAndHour, stepsSinceBoot);
+            db.setTransactionSuccessful();
+            db.endTransaction();
+        } catch (Exception e) {
+            e.printStackTrace();
         }
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: same here.

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 3]
```diff
@@ -194,10 +173,10 @@ public void insertOrReplaceSteps(final long dateAndHour, final int steps) {
                         new String[]{String.valueOf(dateAndHour)});
             }
             db.setTransactionSuccessful();
-        } finally {
             db.endTransaction();
+        } catch (Exception e) {
+            e.printStackTrace();
         }
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: same here for both sqlitedatabase instance and maybe cursor instance should be closed?

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 4]
```diff
@@ -265,43 +244,54 @@ public void logState() {
...
+            if (c.getCount() == 0) {
+                steps = Integer.MIN_VALUE;
+            } else {
+                steps = c.getInt(0);
+            }
+            c.close();
+            return steps;
         } else {
-            steps = c.getInt(0);
+            return c.getInt(0);
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: cursor requires close

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 5]
```diff
@@ -265,43 +244,54 @@ public void logState() {
...
+                sumSteps = 0;
+            } else {
+                c.moveToFirst();
+                sumSteps = c.getInt(0);
+            }
+            c.close();
+            return sumSteps;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return 0;
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: same here. close when getting exception

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 6]
```diff
@@ -265,43 +244,54 @@ public void logState() {
...
-                        COLUMN_DATE_AND_HOUR + " >= ?", new String[]{String.valueOf(start)}, null, null, null);
-        return createChunkedStepCounts(c);
+        try {
+            Cursor c = getReadableDatabase()
+                    .query(TABLE_NAME, new String[]{COLUMN_DATE_AND_HOUR, COLUMN_STEPS},
+                            COLUMN_DATE_AND_HOUR + " >= ?", new String[]{String.valueOf(start)}, null, null, null);
+            return createChunkedStepCounts(c);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: same here.

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 7]
```diff
@@ -311,29 +301,39 @@ public int getSteps(final long start, final long end) {
...
+            SQLiteDatabase db = getWritableDatabase();
+            db.beginTransaction();
+
             updateOrInsertWithoutTransaction(db, DateUtils.getCurrentDateAndHour(), stepsSinceBoot);
             saveLastUpdatedSteps(db, 0);
             db.setTransactionSuccessful();
-        } finally {
             db.endTransaction();
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: plz end transaction

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 8]
```diff
@@ -396,32 +396,33 @@ public int getTodayStep() {
...
             int rows = db.update(TABLE_NAME, values, String.format(COLUMN_DATE_AND_HOUR + " in (%s)", args), dateAndHoursString);
             Logger.log("updated number: " + rows);
             logState();
             db.setTransactionSuccessful();
-        } finally {
             db.endTransaction();
+            Logger.log("update to Recorded: " + dateAndHoursString.toString());
+            db.close();
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: cursor close and end database transaction

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java (hunk 1 above)
Comment (beylerian) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: Alright, we also need to optimize how we open and close. We shouldn't do it every time.

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 9]
```diff
@@ -256,7 +285,7 @@ public int getSteps(final long dateAndHour) {
             c.close();
             return steps;
         } else {
-            return c.getInt(0);
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: Is it alright to change here?

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java [hunk 10]
```diff
@@ -278,15 +310,24 @@ public int getSteps(final long start, final long end) {
...
             e.printStackTrace();
+            if (c != null) {
+                c.close();
+            }
             return 0;
         }
     }
 
     public List<ChunkStepCount> getChunkStepsFrom(final long start) {
+        Cursor c = null;
```
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: no close?

Comment (beylerian) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: `createChunkedStepCounts()` does the close operation

Comment (beylerian) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: I think we should clean up this part entirely later

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java (hunk 9 above)
Comment (beylerian) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: yes actually this case is when `c == null`

File: src/main/java/com/sukesan1984/stepsensorlib/Database.java (hunk 10 above)
Comment (neonankiti) [inline src/main/java/com/sukesan1984/stepsensorlib/Database.java]: got it .

**Comment by neonankiti** [comment]
LGTM +2

**Pull request merged**